promql/engine.go

```diff
@@ -30,6 +30,18 @@ import (
 	"github.com/prometheus/prometheus/util/stats"
 )
 
+const (
+	// The largest SampleValue that can be converted to an int64 without overflow.
+	maxInt64 model.SampleValue = 9223372036854774784
+	// The smallest SampleValue that can be converted to an int64 without underflow.
+	minInt64 model.SampleValue = -9223372036854775808
+)
+
+// convertibleToInt64 returns true if v does not over-/underflow an int64.
+func convertibleToInt64(v model.SampleValue) bool {
+	return v <= maxInt64 && v >= minInt64
```

Comment:
Could this be solved simplar without constants by just:

```
return model.SampleValue(int64(v)) == v
```

Like we do it in the chunk encoding code: https://github.com/prometheus/prometheus/blob/master/storage/local/chunk/delta_helpers.go#L81-L84